server/filters/convert-image-uri.js

```diff
@@ -0,0 +1,75 @@
+
+export default function convertImageUri(originalUri, requiredSize, useIiif, useIiifOrigin) {
+  const imageMap = {
+    wordpress: {
+      root: 'https://wellcomecollection.files.wordpress.com/',
+      iiifRoot: 'https://iiif.wellcomecollection.org/image/wordpress:',
+      iiifOriginRoot: 'https://iiif-origin.wellcomecollection.org/image/wordpress:'
+    },
+    prismic: {
+      root: 'https://prismic-io.s3.amazonaws.com/wellcomecollection/',
+      imigixRoot: 'https://wellcomecollection-prismic.imgix.net',
+      iiifRoot: 'https://iiif.wellcomecollection.org/image/prismic:',
+      iiifOriginRoot: 'https://iiif-origin.wellcomecollection.org/image/prismic:'
+    },
+    miro: {
+      root: 'https://s3-eu-west-1.amazonaws.com/miro-images-public/',
+      imigixRoot: 'https://wellcomecollection-miro-images.imgix.net',
+      iiifRoot: 'https://iiif.wellcomecollection.org/image/',
+      iiifOriginRoot: 'https://iiif-origin.wellcomecollection.org/image/'
+    }
+  };
+
+  const imageSrc = determineSrc(originalUri);
+
+  function determineSrc(url) {
+    if (url.startsWith(imageMap.wordpress.root)) {
+      return 'wordpress';
+    } else if (url.startsWith(imageMap.prismic.root)) {
+      return 'prismic';
+    } else if (url.startsWith(imageMap.miro.root)) {
+      return 'miro';
+    } else {
+      return 'unknown';
+    }
+  }
+
+  function convertPathToWordpressUri(originalUriPath, size) {
+    return originalUriPath + `?w=${size}`;
+  }
+
+  function convertPathToImgixUri(originalUriPath, imgixRoot, size) {
+    return `${imgixRoot}/${originalUriPath}?w=${size}`;
+  }
+
+  function convertPathToIiifUri(originalUriPath, iiifRoot, size) {
+    return `${iiifRoot}${originalUriPath}/full/${size},/0/default.jpg`;
+  }
```

Comment:
Functions in functions mean that we're redeclaring this every time.

Reply:
yep, will move